File: 백준/Silver/1406. 에디터/에디터.js
const filePath = process.platform === 'linux' ? 
'/dev/stdin' : 'example.txt';
const input = require('fs').readFileSync(filePath).toString().trim().split('\n');
const rightStack = [];
let l = input.splice(0, 1);
const str = l[0].split('');


const comand = Number(input[0])

for (let i = 1; i <=comand; i++){
    switch (input[i][0]) { 
        case 'L':
            if (str.length > 0) rightStack.push(str.pop()); 
            break;
        case 'D':
            if (rightStack.length > 0) str.push(rightStack.pop()); 
            break;
        case 'B':
            if (str.length > 0) {
                str.pop();
            }
            break;
        case 'P':
            str.push(input[i][2]);
            break;
    }

}
let answer = str.join('');
answer+=rightStack.reverse().join('');
console.log(answer);

File: 백준/Silver/1406. 에디터/에디터.py
import sys
input = sys.stdin.readline

stack1 = list((input().rstrip()))
stack2 = []
m = int(input())
for _ in range(m):
    order = list(input().split())
    if order[0] == 'L':
        if stack1:
            stack2.append(stack1.pop())

    elif order[0] == 'D':
        if stack2:
            stack1.append(stack2.pop())

    elif order[0] == 'B':
        if stack1:
            stack1.pop()

    elif order[0] == 'P':
        stack1.append(order[1])
stack2 = list(reversed(stack2))
stack1.extend(stack2)

print(''.join(stack1))
